docs/tutorials/creating-your-first-substrate-chain/build.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Building a Custom Runtime Module
+---
+
+We will now modify the `substrate-node-template` to introduce the basic functionality of a Proof Of Existence module.
+
+Open the `substrate-node-template` in your favorite code editor. Then open the file at `runtime/src/template.rs`
+
+```
+substrate-node-template
+|
++-- runtime
+|   |
+|   +-- Cargo.toml
+|   |
+|   +-- build.rs
+|   |
+|   +-- src
+|       |
+|       +-- lib.rs
+|       |
+|       +-- template.rs  <-- Edit this file
+|
++-- scripts
+|
++-- src
+|
++-- ...
+```
+
+You will see some pre-written code which acts as a template for a new runtime module. You can delete the contents of this file since we will start from scratch for full transparency.
+
+At a high level, the a Substrate Runtime Module can be broken down into 5 sections:
+
+```rust
+// 1. Imports
+use support::{decl_module, decl_storage, decl_event,...};
+
+// 2. Module Configuration
+pub trait Trait: system::Trait {...}
+
+// 3. Module Events
+decl_event! {...}
```

Comment:
Shall we follow the convention in [rust book](https://doc.rust-lang.org/book/ch15-05-interior-mutability.html) by using `// --snip--` for code omitting?

Reply:
I used snip in some places, but here I think `...` makes more sense

Is there an example of rustbook using snip like this?

Reply:
cool cool, will update